src/Invictus.Testing/LogicAppsProvider.cs

```diff
@@ -0,0 +1,314 @@
+        /// </summary>
+        /// <param name="timeout">The period to retrieve logic app runs.</param>
+        public LogicAppsProvider WithTimeout(TimeSpan timeout)
+        {
+            _timeout = timeout;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the correlation ID as client tracking when retrieving logic app runs.
+        /// </summary>
+        /// <param name="correlationId">The client tracking of the logic app runs.</param>
+        public LogicAppsProvider WithCorrelationId(string correlationId)
+        {
+            Guard.NotNull(correlationId, nameof(correlationId));
+
+            _hasCorrelationId = true;
+            _correlationId = correlationId;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the tracking property as filter when retrieving logic app runs.
+        /// </summary>
+        /// <param name="trackedPropertyName">The name of the tracked property to filter on.</param>
+        /// <param name="trackedPropertyValue">The value of the tracked property to filter on.</param>
+        public LogicAppsProvider WithTrackedProperty(string trackedPropertyName, string trackedPropertyValue)
+        {
+            Guard.NotNull(trackedPropertyName, nameof(trackedPropertyName));
+            Guard.NotNull(trackedPropertyValue, nameof(trackedPropertyValue));
+
+            _hasTrackedProperty = true;
+            _trackedPropertyName = trackedPropertyName;
+            _trackedPropertyValue = trackedPropertyValue;
+            return this;
+        }
+
+        /// <summary>
+        /// Start polling for a single logic app run corresponding to the previously set filtering criteria.
+        /// </summary>
+        public async Task<LogicAppRun> PollForSingleLogicAppRunAsync()
+        {
+            IEnumerable<LogicAppRun> logicAppRuns = await PollForLogicAppRunsAsync(minimumNumberOfItems: 1);
+            return logicAppRuns.FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Starts polling for a series of logic app runs corresponding to the previously set filtering criteria.
+        /// </summary>
+        public async Task<IEnumerable<LogicAppRun>> PollForLogicAppRunsAsync()
+        {
+            return await PollForLogicAppRunsAsync(minimumNumberOfItems: 1);
+        }
+
+        /// <summary>
+        /// Starts polling for a <paramref name="minimumNumberOfItems"/> corresponding to the previously set filtering criteria.
+        /// </summary>
+        /// <param name="minimumNumberOfItems">The minimum amount of logic app runs to retrieve.</param>
+        public async Task<IEnumerable<LogicAppRun>> PollForLogicAppRunsAsync(int minimumNumberOfItems)
+        {
+            Guard.NotLessThanOrEqualTo(minimumNumberOfItems, 0, nameof(minimumNumberOfItems));
+
+            string amount = minimumNumberOfItems == 1 ? "any" : minimumNumberOfItems.ToString();
```

Comment:
Frankly, this still feels off. Both our methods relying on this provide 1 so we'll always use `any` which is not correct if you ask me.

I'd say that, if you want all runs, you don't provide this property and poll for a given amount of time or so which uses another flow than pulling for one; no? They are different but yet use same exact approach/parameters.

Reply:
Isn't that a problem, since there could be an continuously amount of runs? And we're never sure if we have _all_ runs. I think that's the reason why this was in the initial design.

Reply:
We could use a max timeout but if that's how it worked before then that's it 🤷‍♂️

Reply:
The polling-mechanisms should have a max timeout, so either it returns the given amount of runs or it returns what it found after the max interval expired.
But that should've implemented, so should be OK.

Reply:
Ok, then I guess I have to add two different approaches on this.

Reply:
Ok, @mbraekman , this is now changed to this two-system approach. Can you approve/reject on this new changes? Thanks.